src/jit/TableInl.h

```diff
@@ -0,0 +1,216 @@
+        sljit_emit_op1(compiler, SLJIT_MOV32, SLJIT_MEM1(kContextReg), OffsetOfContextField(tmp2), SLJIT_IMM, ((reinterpret_cast<TableCopy*>(instr->byteCode()))->dstIndex()));
+        sljit_emit_icall(compiler, SLJIT_CALL, SLJIT_ARGS4(VOID, 32, 32, 32, W), SLJIT_IMM, GET_FUNC_ADDR(sljit_sw, copyTable));
+
+        sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R2, 0, SLJIT_R0, 0);
+        sljit_set_label(sljit_emit_cmp(compiler, SLJIT_NOT_EQUAL, SLJIT_R0, 0, SLJIT_IMM, ExecutionContext::NoError), context->trapLabel);
+
+        break;
+    }
+    case TableFillOpcode: {
+        emit3Operands(compiler, instr->operands());
+
+        sljit_emit_op1(compiler, SLJIT_MOV32, SLJIT_MEM1(kContextReg), OffsetOfContextField(tmp1), SLJIT_IMM, ((reinterpret_cast<TableFill*>(instr->byteCode()))->tableIndex()));
+        sljit_emit_icall(compiler, SLJIT_CALL, SLJIT_ARGS4(VOID, 32, 32, 32, W), SLJIT_IMM, GET_FUNC_ADDR(sljit_sw, fillTable));
+
+        sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R2, 0, SLJIT_R0, 0);
+        sljit_set_label(sljit_emit_cmp(compiler, SLJIT_NOT_EQUAL, SLJIT_R0, 0, SLJIT_IMM, ExecutionContext::NoError), context->trapLabel);
+
+        break;
+    }
+    case TableGrowOpcode: {
+        JITArg srcArg;
+
+        operandToArg(instr->operands(), srcArg);
+        sljit_emit_op1(compiler, SLJIT_MOV32, SLJIT_R0, 0, srcArg.arg, srcArg.argw);
+
+        operandToArg(instr->operands() + 1, srcArg);
+        sljit_emit_op1(compiler, SLJIT_MOV_P, SLJIT_R1, 0, srcArg.arg, srcArg.argw);
+
+        sljit_emit_op1(compiler, SLJIT_MOV32, SLJIT_R2, 0, SLJIT_IMM, ((reinterpret_cast<TableGrow*>(instr->byteCode()))->tableIndex()));
+        sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R3, 0, kContextReg, 0);
+        sljit_emit_icall(compiler, SLJIT_CALL, SLJIT_ARGS4(W, P, 32, 32, W), SLJIT_IMM, GET_FUNC_ADDR(sljit_sw, growTable));
+
+        JITArg arg;
+
+        operandToArg(instr->operands() + 2, arg);
+        MOVE_FROM_REG(compiler, SLJIT_MOV32, arg.arg, arg.argw, SLJIT_R0);
+
+        break;
+    }
+    case TableSetOpcode: {
+        JITArg srcArg0;
+        JITArg srcArg1;
+
+        operandToArg(instr->operands(), srcArg0);
+        operandToArg(instr->operands() + 1, srcArg1);
+
+        sljit_emit_op1(compiler, SLJIT_MOV_P, SLJIT_R0, 0, SLJIT_MEM1(kContextReg), OffsetOfContextField(instance));
+        sljit_emit_op1(compiler, SLJIT_MOV_P, SLJIT_R0, 0, SLJIT_MEM1(SLJIT_R0), context->tableStart + ((reinterpret_cast<TableSet*>(instr->byteCode()))->tableIndex() * sizeof(void*)));
+
+        sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R2, 0, SLJIT_IMM, ExecutionContext::OutOfBoundsTableAccessError);
+        sljit_emit_op1(compiler, SLJIT_MOV32, SLJIT_R1, 0, SLJIT_MEM1(SLJIT_R0), JITFieldAccessor::tableSizeOffset());
+        sljit_emit_op1(compiler, SLJIT_MOV32, SLJIT_R3, 0, srcArg0.arg, srcArg0.argw);
+        sljit_set_label(sljit_emit_cmp(compiler, SLJIT_LESS_EQUAL, SLJIT_R1, 0, SLJIT_R3, 0), context->trapLabel);
```

Comment:
Since R1 and R3 are 32 bit, you can use `SLJIT_LESS_EQUAL | SLJIT_32`. Furthermore, loading R3 at this point does not have any advantage over comparing to srcArg0 directly.